### ISSUE TITLE
Array property decorated with ```MinLength``` only will enter an infinite loop

### ISSUE THREAD

**Issue by Pvlerick** [AutoFixture/AutoFixture#1050]
This test will enter an infinite loop:

```csharp
private class TypeWithArrayPropertyWithMinLength
{
    [MinLength(5)]
    public string[] PropertyWithMinLength { get; set; }
}

[Fact]
public void ArrayDecoratedWithMinLengthAttributeShouldPassValidation()
{
    // Arrange
    var sut = new Fixture();

    // Act
    var item = sut.Create<TypeWithArrayPropertyWithMinLength>();

    // Assert
    Validator.ValidateObject(item, new ValidationContext(item), true);
}
```

Funny thing, with ```MaxLength``` only it works fine :-)

**Comment by Pvlerick** [comment]
One more thing: this behavior was not present in 4.3.0 but appeared in 4.4.0

**Comment by zvirja** [comment]
@Pvlerick Good catch! Thanks for reporting the bug!

**Comment by Pvlerick** [comment]
As a matter of fact, it's not an infinite loop, but if there is no ```MaxLength``` property, it will be int.MaxValue, which means that the randomly selected range may request the creation of hundreds of millions of objects :-)

**Comment by zvirja** [comment]
Fixed via #1051
